### PR TITLE
feat(endReached): inverted scrolling

### LINKED ISSUE #80: Inverted scrolling
Creating this ticket to match the PR.

I would like to be able to call the `endReached` callback when I scroll `top`, meaning the callback should be fired when I get the `index` 0.

### PR DESCRIPTION
**Pull request by lbenie**
I'm trying a scrolling chat like behaviour where when we reach the `top` it triggers the `endReached` callback. 

So basically, bottom --> up instead of the current implementation top --> bottom.

I added a debounce in the `listEngine` to prevent multiple calls to `endReached`

closes #80

### CODE REVIEW

File: playground/grouped-inverted.tsx [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+      style={{ height: '350px', width: '400px' }}
+      groupCounts={currentGroupCounts}
+      group={(index: number) => <div>Group {groups[index]}</div>}
+      item={(index: number) => <div>User {index}</div>}
+      initialTopMostItemIndex={45}
+      endReached={index => {
+        console.warn('index', index)
+        loadMore()
+      }}
+      inverted={true}
```
Comment (ysael) [inline playground/grouped-inverted.tsx]: [discussion]: I am wondering if instead of doing `inverted` we should create `startReached ` ?

**Comment by petyosi** [comment]
@ysael @lbenie  I think startReached would be a better choice, indeed. An inverted flag sounds like it will offer much more than that (like starting the list from the bottom, etc.).

Question - why don't you use the the [rangeChange event](https://virtuoso.dev/range-change-callback/)? You can check the startIndex for zero. I might be missing something, though. 

Thank you both for the participation!

**Comment by lbenie** [comment]
> @ysael @lbenie I think startReached would be a better choice, indeed. An inverted flag sounds like it will offer much more than that (like starting the list from the bottom, etc.).
> 
> Question - why don't you use the the [rangeChange event](https://virtuoso.dev/range-change-callback/)? You can check the startIndex for zero. I might be missing something, though.
> 
> Thank you both for the participation!

@petyosi Thanks for responding, after much more thoughts the `rangeChanged` event can work for this particular case. Although, the inverted list is something that could be useful for a consumer though.
What are your thoughts on this?

If @ysael PR is merged and released, we can continue our work 👍

**Comment by petyosi** [comment]
I merged the PR in v0.14.0 - please give it a try. If it works for you, let me know. 

Cheers,

**Comment by lbenie** [comment]
> I merged the PR in v0.14.0 - please give it a try. If it works for you, let me know.
> 
> Cheers,

Works thanks!

**Pull request closed without merging**